test/project_types/script/layers/infrastructure/languages/assemblyscript_task_runner_test.rb

```diff
@@ -195,4 +220,125 @@ def self.it_triggers_compilation_process
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is above minimum" do
+      describe "when node version is below minimum" do
+        it "should raise error" do
+          ctx.expects(:capture2e)
+            .with("npm", "--version")
+            .returns([ABOVE_NPM_VERSION, mock(success?: true)])
+          ctx.expects(:capture2e)
+            .with("node", "--version")
+            .returns([BELOW_NODE_VERSION, mock(success?: true)])
+          assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
+            subject
+          end
+        end
+      end
+
+      describe "when node version is above minimum" do
+        it "should install successfully" do
+          ctx.expects(:capture2e)
+            .with("npm", "--version")
+            .returns([ABOVE_NPM_VERSION, mock(success?: true)])
+          ctx.expects(:capture2e)
+            .with("node", "--version")
+            .returns([ABOVE_NODE_VERSION, mock(success?: true)])
+          subject
+        end
+      end
+
+      describe "when node version is the exact version" do
+        it "should install successfully" do
+          ctx.expects(:capture2e)
+            .with("npm", "--version")
+            .returns([ABOVE_NPM_VERSION, mock(success?: true)])
+          ctx.expects(:capture2e)
+            .with("node", "--version")
+            .returns([EXACT_NODE_VERSION, mock(success?: true)])
+          subject
+        end
+      end
+    end
+
+    describe "when npm version is exactly the version" do
+      describe "when node version is below minimum" do
+        it "should raise error" do
+          ctx.expects(:capture2e)
+            .with("npm", "--version")
+            .returns([EXACT_NPM_VERSION, mock(success?: true)])
+          ctx.expects(:capture2e)
+            .with("node", "--version")
+            .returns([BELOW_NODE_VERSION, mock(success?: true)])
+          assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
+            subject
+          end
+        end
+      end
+
+      describe "when node version is above minimum" do
+        it "should install successfully" do
+          ctx.expects(:capture2e)
+            .with("npm", "--version")
+            .returns([EXACT_NPM_VERSION, mock(success?: true)])
+          ctx.expects(:capture2e)
+            .with("node", "--version")
+            .returns([ABOVE_NODE_VERSION, mock(success?: true)])
+          subject
+        end
+      end
+
+      describe "when node version is the exact version" do
+        it "should install successfully" do
+          ctx.expects(:capture2e)
+            .with("npm", "--version")
+            .returns([EXACT_NPM_VERSION, mock(success?: true)])
+          ctx.expects(:capture2e)
+            .with("node", "--version")
+            .returns([EXACT_NODE_VERSION, mock(success?: true)])
+          subject
+        end
+      end
+    end
```

Comment:
In cases like these where you collect related tests, you can break out common clauses into a helper so that we avoid repeating them. If you pull the helper into the top layer of the test (meaning in the top level context of the  `describe Script::Layers::Infrastructure::Languages::AssemblyScriptTaskRunner do` block) then you can do these with your other describe blocks too.

```suggestion
    describe "when npm version is exactly the version" do
      def stub_tool_versions(npm:, node:)
        ctx.stubs(:capture2e)
          .with("npm", "--version")
          .returns([npm, mock(success?: true)])
        ctx.stubs(:capture2e)
          .with("node", "--version")
          .returns([node, mock(success?: true)])
      end
    
      describe "when node version is below minimum" do
        it "should raise error" do
          stub_tool_versions(npm: EXACT_NPM_VERSION, node: BELOW_NODE_VERSION)
          assert_raises Script::Layers::Infrastructure::Errors::MissingDependencyVersionError do
            subject
          end
        end
      end

      describe "when node version is above minimum" do
        it "should install successfully" do
          stub_tool_versions(npm: EXACT_NPM_VERSION, node: ABOVE_NODE_VERSION)
          subject
        end
      end

      describe "when node version is the exact version" do
        it "should install successfully" do
          stub_tool_versions(npm: EXACT_NPM_VERSION, node: EXACT_NODE_VERSION)
          subject
        end
      end
    end
```